File: app/src/main/java/com/bumie/nearme_/Datum.java
package com.bumie.nearme_;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Datum {

    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("subType")
    @Expose
    private String subType;
    @SerializedName("id")
    @Expose
    private String id;

    @SerializedName("geoCode")
    @Expose
    private GeoCode geoCode;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("category")
    @Expose
    private String category;
    @SerializedName("rank")
    @Expose
    private Integer rank;
    @SerializedName("tags")
    @Expose
    private List<String> tags = null;

    public Datum(String name, String category) {
        this.name = name;
        this.category = category;
        //this.mLatitude = latitude;
        //this.mLongitude = longitude;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getSubType() {
        return subType;
    }

    public void setSubType(String subType) {
        this.subType = subType;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public GeoCode getGeoCode() {
        return geoCode;
    }

    public void setGeoCode(GeoCode geoCode) {
        this.geoCode = geoCode;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public Integer getRank() {
        return rank;
    }

    public void setRank(Integer rank) {
        this.rank = rank;
    }

    public List<String> getTags() {
        return tags;
    }

    public void setTags(List<String> tags) {
        this.tags = tags;
    }

}

File: app/src/main/java/com/bumie/nearme_/Datum.kt
package com.bumie.nearme_

import android.os.Parcel
import android.os.Parcelable

data class Datum(
    var name:String,
    var category:String,
    var type:String,
    var subType:String,
  //  var geoCode: GeoCode,
    var rank:Int,
    var tags:List<String>): Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString()!!,
        parcel.readString()!!,
        parcel.readString()!!,
        parcel.readString()!!,
       // parcel.readString()!!,
        parcel.readInt(),
        parcel.createStringArrayList()!!
    )

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, p1: Int) {
        parcel.writeString(name)
        parcel.writeString(type)
        parcel.writeString(subType)
        parcel.writeString(category)
       // parcel.writeValue(geoCode)
        parcel.writeInt(rank)
        parcel.writeStringList(tags)
    }

    companion object CREATOR : Parcelable.Creator<Datum> {
        override fun createFromParcel(parcel: Parcel): Datum {
            return Datum(parcel)
        }

        override fun newArray(size: Int): Array<Datum?> {
            return arrayOfNulls(size)
        }
    }


}
